BHoM_Adapter/AdapterActions/Pull.cs

```diff
@@ -42,6 +42,33 @@ public abstract partial class BHoMAdapter
         /* These methods represent Actions that the Adapter can complete. 
            They are publicly available in the UI as individual components, e.g. in Grasshopper, under BHoM/Adapters tab. */
 
+        [Description("Performs a set up for the Request input of the Pull Action.")]
+        public virtual bool SetupPullRequest(object request, out IRequest actualRequest)
+        {
+            // Always assure there is a Request. Allow to input a Type to generate a FilterRequest.
+            actualRequest = null;
+
+            if (request is Type)
```

Comment:
this is now jsut details, but as I commented below anyway, would change the order of these again, mostly just for clarity to:

1. check null
2. check if request
3. check if type

Mostly as an order of "importance" and for general validation purpose. as in, first check for null, then check if it is the exact thing you are after, then check for any additional possibility (in this case the type)

Reply:
Sounds good, done.